src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1544,6 +1544,161 @@ private function insert_html_element( $token ) {
 	 * HTML Specification Helpers
 	 */
 
+	/**
+	 * Returns whether a given element is an HTML tag name.
+	 *
+	 * @todo Verify this list.
```

Comment:
In deeper inspection here I feel like this is the wrong approach. We should be able to instead determine if we're in foreign content and apply the rules there accordingly. That way we don't have to keep a list of HTML elements updated _and_ we don't have to worry about conflating elements with the same name of HTML elements with foreign elements, e.g. `TITLE` inside an SVG.

So I think more important now is getting foreign content detection in place. I've started working on this in #6006.

It may be the case that this relies on the HTML Processor because the rules get complicated with MathML and foriegn content integration points.

Reply:
obviously after exploring #6006 there's no easy way to infer the outside context when creating a tag in isolation. this leads me to think that we need a good way to communicate this to developers.

maybe instead of passing `'self-closing'` we could have people pass `'xml-empty-tag'` or `'empty-tag-in-foreign-content'`. I'd like to communicate that this isn't for `IMG` or any HTML tag.

the trickiest part I've come to realize is that we could have an `SVG IMG` tag as well, which could adopt the self-closing identity by nature of being inside foreign content, and that would be meaningful.

I don't see this being often used, so I feel comfortable making it a bit awkward. it seems incredibly unlikely to be common that someone is intentionally creating an empty XML element inside foreign content.